cli/utils.go

```diff
@@ -107,7 +105,7 @@ func parseLabels(inputLabels []string) (models.LabelSet, error) {
 	labelSet := make(models.LabelSet, len(inputLabels))
 
 	for _, l := range inputLabels {
-		matcher, err := labels.ParseMatcher(l)
```

Comment:
Sure! Fwiw, I consider this behaviour broken:

> We use the same parser for labels as for matchers because they follow the same grammar

I wrote in another comment about fixing it in future. Labels cannot have `{` or trailing commas, so the grammar is not the same.

Reply:
Sure! Fwiw, I consider this behaviour broken:

> We use the same parser for labels as for matchers because they follow the same grammar

I wrote in another comment about fixing it in future. Labels cannot start with `{`, end with `}` or having trailing commas, so the grammar is not the same.

Reply:
I think this comment is irrelevant now as we are going to remove the `parseLabels` as per https://github.com/prometheus/alertmanager/pull/3483#discussion_r1334205799. I can inline the comment where the function used to be though, if that's helpful?